src/components/ContentNode/TabNavigator.vue

```diff
@@ -0,0 +1,90 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2021 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+<template>
+  <div
+    class="TabNavigator"
+    :class="[{ 'tabs--vertical': vertical }]"
+  >
+    <Tabnav v-model="current" v-bind="{ position, vertical }">
+      <TabnavItem v-for="(item, index) in items" :key="index" :value="index">
+        {{ item.title }}
+      </TabnavItem>
+    </Tabnav>
+    <div class="content">
+      <ContentNode :content="currentContent" />
+    </div>
+  </div>
+</template>
+
+<script>
+import Tabnav from 'docc-render/components/Tabnav.vue';
+import TabnavItem from 'docc-render/components/TabnavItem.vue';
+
+/**
+ * Tab navigation component, that renders `ContentNode`,
+ * with ability to align the tabs horizontally and vertically to the start/center/end of the line.
+ * It can also flip the navigation, so it renders content first, navigation second,
+ * in both horizontal and vertical mode.
+ */
+export default {
+  name: 'TabNavigator',
+  components: {
+    TabnavItem,
+    Tabnav,
+    ContentNode: () => import('docc-render/components/ContentNode.vue'),
```

Comment:
I recall you asking me about different ways to solve this problem with recursive use of the `ContentNode` component, and I remember now how we have done this in the past in similar situations:

In `ContentNode`, where there are places where certain block elements can render other block elements, we have a recursive function that utilizes the slots to recursively render the nested children of a given block, and that avoids the need of using dynamic imports like this.

This situation might be _slightly_ different, because you have an array of blocks as opposed to a single block, but I _think_ it could possibly be solved the same way depending on the API of `$slots`.

Reply:
You mean the `renderChildren` function?

How would I know which child is active and should be rendered in that case? I would have to move/expose that logic out of the TabNavigator and into to the ContentNode, which feels unnecessary.

Reply:
I think so, yeah. I see what you mean—to do it that way, you would have to pre-render all the children first and then decide what to hide/show.

Anyways, that was the only similar thing I could think of. I don't think there's necessarily an issue with the dynamic import, as long as it works the way it should.